newVersion.py

```diff
@@ -7,33 +7,55 @@
 import os
 import textwrap
 
+# ===================================================================================
+# Begin editable globals.
+# This section contains global variables with values that might change in the future.
+# ===================================================================================
+
 # iTwin base version to search for. 3.0.x for now.
 itwin_base_version = "3\\.0\\."
 # iTwin Mobile SDK base version to search for. 0.10.x for now.
 mobile_base_version = "0\\.10\\."
 # Subdirectory under mobile-samples of react-app.
 react_app_subdir = 'cross-platform/react-app'
-# The relative paths to the elements in the dirs array
-relative_dirs = [
-    '.',
-    '../mobile-sdk-core',
-    '../mobile-ui-react',
-    '../mobile-samples',
-]
 # The names of the sample apps
 sample_names = [
     'MobileStarter',
     'SwiftUIStarter',
     'ThirdPartyAuth',
 ]
-# Indices of the elements of the dirs array
-mobile_sdk_ios_dir_index = 0
-mobile_sdk_core_dir_index = 1
-mobile_ui_react_dir_index = 2
-mobile_samples_dir_index = 3
 
-def get_executing_directory():
-    return os.path.dirname(os.path.realpath(__file__))
+# ===================================================================================
+# End editable globals.
+# ===================================================================================
+
+class MobileSdkDirs:
+    def __init__(self):
+        executing_dir = os.path.dirname(os.path.realpath(__file__))
+        # The relative paths to the iTwin Mobile SDK repository directories.
+        # Since these are directly tied to member properties on this class, they cannot
+        # be in the editable globals section above.
+        relative_dirs = [
+            '.',
+            '../mobile-sdk-core',
+            '../mobile-ui-react',
+            '../mobile-samples',
+        ]
+        def build_dir(dir_name):
+            return os.path.realpath(os.path.join(executing_dir, dir_name))
+        self.dirs = []
+        for relative_dir in relative_dirs:
+            self.dirs.append(build_dir(relative_dir))
+        self.sdk_ios = self.dirs[0]
+        self.sdk_core = self.dirs[1]
+        self.ui_react = self.dirs[2]
+        self.samples = self.dirs[3]
+
+    def __iter__(self):
+        for dir in self.dirs:
```

Comment:
Can we just `yield self.dirs`? Or we can yield the 4 dirs and remove self.dirs completely.

Reply:
I tried that, and it resulted in the whole array being the only hit when iterating, instead of iterating through the items in the array. In other words, for dir in sdk_dirs: would have one and only one dir, and that dir was the array. I could get rid of the __iter__ entirely, and use for dir in sdk_dirs.dirs:, but I felt that this was much better.